src/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -276,7 +278,23 @@ public ReadOnlySequence<T> Slice(long start)
         }
 
         /// <inheritdoc />
-        public override string ToString() => string.Format("System.Buffers.ReadOnlySequence<{0}>[{1}]", typeof(T).Name, Length);
+        public override string ToString()
+        {
+            if (typeof(T) == typeof(char) && Length < int.MaxValue)
+            {
+                return string.Create((int)Length, this, (span, sequence) =>
```

Comment:
What if the underlying Memory is backed by a string? Is there a way we can special case that?

Reply:
Will do